hnn_core/tests/test_gui.py

```diff
@@ -1,8 +1,198 @@
 # Authors: Huzi Cheng <hzcheng15@icloud.com>
-from hnn_core.gui.gui import run_hnn_gui
+import os.path as op
 
+import hnn_core
+import matplotlib
+import numpy as np
+import pytest
+from hnn_core import Dipole, Network, Params
+from hnn_core.gui import HNNGUI
+from hnn_core.gui.gui import _init_network_from_widgets
+from hnn_core.network import pick_connection
+from hnn_core.parallel_backends import requires_mpi4py, requires_psutil
 
-def test_run_gui():
-    """Test if main gui function gives proper ipywidget"""
-    gui = run_hnn_gui()
-    assert gui is not None
+matplotlib.use('agg')
+
+
+def test_gui_load_params():
+    """Test if gui loads default parameters properly"""
+    gui = HNNGUI()
+
+    assert isinstance(gui.params, Params)
+
+    print(gui.params)
+    print(gui.params['L2Pyr*'])
+
+
+def test_gui_upload_params():
+    """Test if gui handles uploaded parameters correctly"""
+    gui = HNNGUI()
+    _ = gui.compose()
+
+    params_name = 'default.json'
+    hnn_core_root = op.join(op.dirname(hnn_core.__file__))
+    params_fname = op.join(hnn_core_root, 'param', params_name)
+
+    content = b""
+    with open(params_fname, "rb") as f:
+        for line in f:
+            content += line
+    uploaded_value = {
+        params_name: {
+            'metadata': {
+                'name': params_name,
+                'type': 'application/json',
+                'size': len(content),
+            },
+            'content': content
+        }
+    }
+
+    # change the default loaded parameters
+    original_drive_count = len(gui.drive_widgets)
+    assert original_drive_count > 0
+    gui.delete_drive_button.click()
+    assert len(gui.drive_widgets) == 0
+
+    original_tstop = gui.widget_tstop.value
+    gui.widget_tstop.value = 1
+
+    original_tstep = gui.widget_dt.value
+    gui.widget_dt.value = 1
+
+    # manually send uploaded content
+    gui.load_button.set_trait('value', uploaded_value)
+
+    # check if parameter is reloaded.
+    assert gui.widget_tstop.value == original_tstop
+    assert gui.widget_dt.value == original_tstep
+    assert len(gui.drive_widgets) == original_drive_count
+
+
+def test_gui_change_connectivity():
+    """Test if GUI properly changes cell connectivity parameters.
+    """
```

Comment:
The quotes should be in the previous line. Fix everywhere else. One line docstrings should really be one line.